### ISSUE TITLE
model.optimize's default for objective_direction is problematic with optlang

### ISSUE THREAD

**Issue by cdiener** [opencobra/cobrapy#372]
So `model.optimize` has an argument `objective_direction` that defaults to "maximize". However, optlang manages the objective direction as an attribute of its objective. However, calling `model.optimize()` will ignore this attribute and set it to a maximization. Due to this:

```
model.objective = interface.Objective(expr, direction="min")
model.objective()
```

will cause one of two behaviors:

- it will maximizes instead of minimizing giving an incorrect optimum
- it will raise a cryptic error from the low level interface if the objective can not be maximized, like "Cplex Error 5001: objective is not convex" or something like that (optlang does not catch low level errors in that case)

I think a better default would be `objective_direction=None` and setting maximization only for the legacy interface. This would respect `model.solver.objective.direction`.

**Comment by hredestig** [comment]
Just got that error as well, indeed cryptic, good proposal

**Comment by hredestig** [comment]
fixed according to proposal #411
